### PR TITLE
WT-10226 Remove WT_PAGE_DEL::previous_ref_state

### PR DESCRIPTION
**Pull request by tod-johnson-mongodb**
It always == WT_REF_DISK so it is not needed.

### CODE REVIEW

File: src/include/btmem.h [hunk 1]
```diff
@@ -1047,10 +1047,10 @@ struct __wt_page_deleted {
     wt_shared volatile uint8_t prepare_state;
 
     /*
-     * The previous state of the WT_REF; if the fast-truncate transaction is rolled back without the
-     * page first being instantiated, this is the state to which the WT_REF returns.
+     * The previous state of the WT_REF is always set to WT_REF_DISK and is not stored. If the
```
Comment (luke-pearson) [inline src/include/btmem.h:1050]: Delete this comment.

File: src/include/btree_inline.h [hunk 2]
```diff
@@ -1566,7 +1566,7 @@ __wt_ref_addr_copy(WT_SESSION_IMPL *session, WT_REF *ref, WT_ADDR_COPY *copy)
             copy->del.txnid = WT_TXN_NONE;
             copy->del.timestamp = copy->del.durable_timestamp = WT_TS_NONE;
             copy->del.prepare_state = 0;
-            copy->del.previous_ref_state = WT_REF_DISK;
+            /* copy_del->previous_ref_state, not stored, = WT_REF_DISK as always. */
```
Comment (luke-pearson) [inline src/include/btree_inline.h:1569]: Remove this.

File: src/include/cell_inline.h [hunk 3]
```diff
@@ -893,9 +893,12 @@ __wt_cell_unpack_safe(WT_SESSION_IMPL *session, const WT_PAGE_HEADER *dsk, WT_CE
         WT_RET(__wt_vunpack_uint(&p, end == NULL ? 0 : WT_PTRDIFF(end, p), &page_del->timestamp));
         WT_RET(__wt_vunpack_uint(
           &p, end == NULL ? 0 : WT_PTRDIFF(end, p), &page_del->durable_timestamp));
-        page_del->prepare_state = 0;                /* No prepare can have been in progress. */
-        page_del->previous_ref_state = WT_REF_DISK; /* The leaf page is on disk. */
-        page_del->committed = true;                 /* There is no running transaction. */
+        page_del->prepare_state = 0; /* No prepare can have been in progress. */
+        /*
```
Comment (luke-pearson) [inline src/include/cell_inline.h:897]: Remove this.

File: src/btree/bt_delete.c [hunk 4]
```diff
@@ -194,9 +194,11 @@ __wti_delete_page(WT_SESSION_IMPL *session, WT_REF *ref, bool *skipp)
      */
     WT_ERR(__wt_page_parent_modify_set(session, ref, false));
 
-    /* Allocate and initialize the page-deleted structure. */
+    /*
+     * Allocate and initialize the page-deleted structure. Note previous_ref_state, not stored, ==
```
Comment (luke-pearson) [inline src/btree/bt_delete.c:198]: Remove comment about previous_ref_state = WT_REF_DISK here.

File: src/btree/bt_vrfy_dsk.c [hunk 5]
```diff
@@ -332,7 +332,7 @@ __verify_dsk_addr_page_del(WT_SESSION_IMPL *session, WT_CELL_UNPACK_ADDR *unpack
      * evicted.
      */
     WT_ASSERT(session, unpack->page_del.prepare_state == 0);
-    WT_ASSERT(session, unpack->page_del.previous_ref_state == WT_REF_DISK);
+    /* unpack->page_del->previous_ref_state, not stored, == WT_REF_DISK as always. */
```
Comment (luke-pearson) [inline src/btree/bt_vrfy_dsk.c:335]: Remove this.

File: src/btree/bt_delete.c [hunk 6]
```diff
@@ -267,7 +269,7 @@ __wt_delete_page_rollback(WT_SESSION_IMPL *session, WT_REF *ref)
      * the update list to be null to be conservative.
      */
     if (current_state == WT_REF_DELETED) {
-        current_state = ref->page_del->previous_ref_state;
```
Comment (luke-pearson) [inline src/btree/bt_delete.c:270]: Add a comment here:

When fast truncate succeeds it moves a WT_REF from WT_REF_DISK to WT_REF_DELETED, given that we know that the reverse operation of this is to return the state to WT_REF_DISK.

File: src/include/btmem.h (hunk 1 above)
Comment (tod-johnson-mongodb) [reply src/include/btmem.h:1050]: Done

File: src/include/btree_inline.h (hunk 2 above)
Comment (tod-johnson-mongodb) [reply src/include/btree_inline.h:1569]: Done

File: src/include/cell_inline.h (hunk 3 above)
Comment (tod-johnson-mongodb) [reply src/include/cell_inline.h:897]: Done

File: src/btree/bt_delete.c (hunk 4 above)
Comment (tod-johnson-mongodb) [reply src/btree/bt_delete.c:198]: Done

File: src/btree/bt_vrfy_dsk.c (hunk 5 above)
Comment (tod-johnson-mongodb) [reply src/btree/bt_vrfy_dsk.c:335]: Done

**Review by luke-pearson — APPROVED** [review]
LGTM.

**Pull request merged**